arangod/Cluster/ClusterTrxMethods.cpp

```diff
@@ -307,33 +334,22 @@ Future<Result> beginTransactionOnLeaders(TransactionState& state,
   Result res;
   if (leaders.empty()) {
     return res;
-  }
-
-  std::vector<Future<network::Response>> requests;
-  for (ServerID const& leader : leaders) {
+  } else if (leaders.size() == 1) { // shortcut
+    ServerID const& leader = leaders.back();
     if (state.knowsServer(leader)) {
-      continue;  // already send a begin transaction there
+      return res;  // already send a begin transaction there
     }
     state.addKnownServer(leader);
-    requests.emplace_back(::beginTransactionRequest(state, leader));
-  }
-
-  if (requests.empty()) {
-    return res;
+    
+    const TransactionId tid = state.id().child();
+    return ::beginTransactionRequest(state, leader)
+    .thenValue([tid](network::Response&& resp) {
+      return ::checkTransactionResult(tid, transaction::Status::RUNNING, resp);
+    });
   }
-
-  const TransactionId tid = state.id().child();
-  return futures::collectAll(requests).thenValue(
-      [=](std::vector<Try<network::Response>>&& responses) -> Result {
-        for (Try<arangodb::network::Response> const& tryRes : responses) {
-          network::Response const& resp = tryRes.get();  // throws exceptions upwards
-          Result res = ::checkTransactionResult(tid, transaction::Status::RUNNING, resp);
-          if (res.fail()) {  // remove follower from all collections
-            return res;
-          }
-        }
-        return Result();  // all good
-      });
+  auto servers = std::make_shared<std::vector<ServerID>>(leaders);
+  std::sort(servers->begin(), servers->end(), std::greater<ServerID>());
```

Comment:
You are sorting the servers here and also sorting them on the getRelevantServers function above.

If I am not mistaken this code place here is the critical one, the other can work with non-sortedness of Servers.
Can you please verify if I am off here, or if we can get away with the sorting here alone.

Reply:
no, this does not use getRelevantServers()